### PR TITLE
NCL-901 Use layered image structure. Add docker image for pnc-build-a…

### PR DESCRIPTION
**Pull request by rnc**
…gent.

### CODE REVIEW

**Comment by slaskawi** [comment]
For me it looks good :+1:

File: docker-environment-driver/src/main/docker/pnc-build-agent/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+RUN mv /usr/share/maven/conf/settings.xml.backup /usr/share/maven/conf/settings.xml
+
+EXPOSE 8080
+
+# Fix maven settings location
+
+
+# Prepare script to start BuildAgent, SSH daemon and ip-tables
+RUN echo -e "/usr/bin/setup-proxy-server.sh & \n /usr/bin/isolate-with-iptables.sh & \n /usr/sbin/sshd -D & \n java -jar /usr/local/build-agent-0.1-SNAPSHOT-jar-with-dependencies.jar" > /usr/bin/startPrograms.sh
+
```
Comment (matejonnet) [inline docker-environment-driver/src/main/docker/pnc-build-agent/Dockerfile]: @jbartece do we still need ssh server running in a docker container?

Comment (jbartece) [inline docker-environment-driver/src/main/docker/pnc-build-agent/Dockerfile]: Good point Matej. It was removed from the Docker driver and it is not needed now.

Comment (rnc) [inline docker-environment-driver/src/main/docker/pnc-build-agent/Dockerfile]: That can be removed although it was actually extremely useful to work out what was going wrong with the build agent driver putting it in the docker image

File: docker-environment-driver/src/main/docker/pnc-build-agent/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+RUN /usr/share/maven/bin/mvn -q org.apache.maven.plugins:maven-dependency-plugin:2.10:get -DrepoUrl=https://repository.jboss.org/nexus/content/repositories/snapshots -Dartifact=org.jboss.pnc:build-agent:0.1-SNAPSHOT:jar:jar-with-dependencies -Ddest=/usr/local/build-agent-0.1-SNAPSHOT-jar-with-dependencies.jar
+RUN mv /usr/share/maven/conf/settings.xml.backup /usr/share/maven/conf/settings.xml
+
+EXPOSE 8080
+
+# Fix maven settings location
+
+
+# Prepare script to start BuildAgent, SSH daemon and ip-tables
+RUN echo -e "/usr/bin/setup-proxy-server.sh & \n /usr/bin/isolate-with-iptables.sh & \n /usr/sbin/sshd -D & \n java -jar /usr/local/build-agent-0.1-SNAPSHOT-jar-with-dependencies.jar" > /usr/bin/startPrograms.sh
```
Comment (slaskawi) [inline docker-environment-driver/src/main/docker/pnc-build-agent/Dockerfile]: Could you please add an option to BuildAgent to run in `/tmp` directory? I need a fixed path for build.

**Pull request merged**